File: client/types/plans.ts
export interface Assignment {
  assignment_id: string;
  task_id: string;
  task_name: string;
  task_mode: string;
  day_index: number | null;
  status: string;
  priority_order: number | null;
  initial_level_name?: string | null;
  current_level?: string | null;
}

export interface TodayTasksResponse {
  assignments: Assignment[];
  any_escalated: boolean;
}

export interface Plan {
  plan_id: string;
  plan_name: string;
  start_date: string | null;
  end_date: string | null;
  status: "draft" | "approved" | "archived";
  goals: string | null;
  assignments: Assignment[];
}

export interface PlanRevisionEntry {
  id: string;
  action: string;
  actor_role: string;
  change_summary: string | null;
  created_at: string;
}

File: server/app/schemas/plans.py
from pydantic import BaseModel
from typing import Optional


class GeneratePlanRequest(BaseModel):
    patient_id: str
    baseline_level: str = "easy"


class AssignmentOut(BaseModel):
    assignment_id: str
    task_id: str
    task_name: str
    task_mode: str
    day_index: int | None
    status: str
    priority_order: int | None


class PlanOut(BaseModel):
    plan_id: str
    plan_name: str
    start_date: str | None
    end_date: str | None
    status: str
    goals: str | None
    assignments: list[AssignmentOut]


class AddTaskRequest(BaseModel):
    task_id: str
    day_index: int
    priority_order: int = 0


class UpdateAssignmentRequest(BaseModel):
    day_index: int | None = None
    status: str | None = None


class TaskForDefectOut(BaseModel):
    task_id: str
    name: str
    task_mode: str
    type: str


class PlanRevisionEntryOut(BaseModel):
    id: str
    action: str
    actor_role: str
    change_summary: Optional[str]
    created_at: str
